src/Slices/DesiredState/UI/Page.test.tsx

```diff
@@ -462,3 +464,108 @@ test("DesiredStatesView shows CompileWidget", async () => {
   render(component);
   expect(screen.getByRole("generic", { name: "CompileWidget" })).toBeVisible();
 });
+
+describe("DeleteModal ", () => {
+  it("Shows form when clicking on modal button", async () => {
+    const { component, apiHelper } = setup();
+    render(component);
+
+    apiHelper.resolve(204);
+
+    await act(async () => {
+      await apiHelper.resolve(Either.right(DesiredStateVersions.response));
+    });
+    const rows = await screen.findAllByRole("row", {
+      name: "DesiredStates Table Row",
+    });
+
+    await userEvent.click(
+      within(rows[0]).getByRole("button", {
+        name: "Actions",
+      })
+    );
+
+    await userEvent.click(
+      within(screen.getByRole("menu", { name: "Actions" })).getByText("Delete")
+    );
+    expect(
+      await screen.findByText("Are you sure you want to delete version 9?")
+    ).toBeVisible();
+    expect(await screen.findByText("Yes")).toBeVisible();
+    expect(await screen.findByText("No")).toBeVisible();
+  });
+  it("Closes modal when cancelled(both cancel buttons scenario)", async () => {
+    const { component, apiHelper } = setup();
+    render(component);
+
+    apiHelper.resolve(204);
+
+    await act(async () => {
+      await apiHelper.resolve(Either.right(DesiredStateVersions.response));
+    });
+    const rows = await screen.findAllByRole("row", {
+      name: "DesiredStates Table Row",
+    });
+
+    //close by "no" button scenario
```

Comment:
you mean clicking outside the container of the modal?

Reply:
haaa okay, the button with "no" written in it xD

Reply:
exactly :D